src/corona_hakab_model/matrix_analysis/MatrixAnalysis.py

```diff
@@ -0,0 +1,135 @@
+from argparse import ArgumentParser
+from matplotlib import pyplot as plt
+from pathlib import Path
+import numpy as np
+import csv
+
+from generation.matrix_generator import MatrixData
+
+
+class MatrixAnalyzer:
+    """
+    A module for analyzing and plotting statistics of the simulation matrix.
+    """
+
+    EXPORT_HISTOGRAM_DIR = "../../output/matrix_analysis/histogram_analysis/"
+    EXPORT_HISTOGRAM_FILE = "histogram_connection_type_{}.csv"
+    EXPORT_HISTOGRAM_IMG = "histogram_connection_type_{}.png"
+    EXPORT_MATRIX_DIR = "../../output/matrix_analysis/raw_matrices/"
+    EXPORT_MATRIX_FILE = "raw_matrix_connection_type_{}.csv"
+
+    def __init__(self, matrix_data_path=None):
+        self.matrix_data = MatrixData()
+        self.matrix_data.import_matrix_data_as_scipy_sparse(matrix_data_path)
+        self.matrix_data.depth = len(self.matrix_data.matrix)
+        self._convert_to_csr()
+
+        Path(self.EXPORT_HISTOGRAM_DIR).mkdir(parents=True, exist_ok=True)
+        Path(self.EXPORT_MATRIX_DIR).mkdir(parents=True, exist_ok=True)
+
+    def export_raw_matrices_to_csv(self):
+        """
+        run over all types of connections and export each of them to a csv file
+        """
+        for connection_type in range(self.matrix_data.depth):
+            self.export_matrix_to_csv(connection_type)
+        # all connections combined
+        self.export_matrix_to_csv()
+
+    def export_matrix_to_csv(self, connection_type=None):
+        if connection_type is None:
+            connection_matrix = sum(self.matrix_data.matrix)
+            connection_type = "all"
+        else:
+            connection_matrix = self.matrix_data.matrix[connection_type]
+
+        # get the indices and values of all non-zero values in the matrix
+        (connection, agent), value = connection_matrix.nonzero(), connection_matrix.data
+        connections = map(lambda x, y, z: [x, y, z], agent, connection, value)
+        connections = sorted(connections)
+
+        headers = ['agent', 'connection', 'value']
+        export_file_path = self.EXPORT_MATRIX_DIR + self.EXPORT_MATRIX_FILE.format(connection_type)
+        with open(export_file_path, 'w', newline='') as matrix_file:
+            csv_writer = csv.writer(matrix_file, delimiter=',')
+            csv_writer.writerows([headers]+connections)
+
+    def analyze_histograms(self):
+        """
+        run over all types of connections and analyze the connection histogram
+        """
+        for connection_type in range(self.matrix_data.depth):
+            self.analyze_connection_histogram(connection_type)
+        self.analyze_connection_histogram()
```

Comment:
Please add comment that explains why you need this empty call

Reply:
תוקן